### ISSUE TITLE
Run git-deps transparently/automatically

### ISSUE THREAD

**Issue by rplevy-draker** [tobyhede/lein-git-deps#5]
Hi Toby,

I haven't studied the source much yet, but is there any way we could make git-deps run similarly to how deps runs?  The real advantage to this is that deps runs transparently and is hardly ever necessary to run manually. Similarly git deps could run prior to any lein command that might require it. Presumably the same situations in which the check for deps occurs.

Perhaps this automatic running of git-deps should be an option to enable or disable because not everyone would want it, and perhaps no one would want it on all of the time.  But for me at least, the typical use case is that I have a dependency used by multiple projects, so when there is a change in the dependency, I need to go into every project and lein git-deps.

The lein-git-deps plugin is working great by the way. We have it checking out into the lein ./checkouts directory.

**Comment by pchalasani** [comment]
I agree, it would be great if I simply put the appropriate `:git-dependencies` and `:plugins` into my `project.clj`, and then when I run `lein uberjar`, it should: (a) git clone/pull the repo from github, (b) do `lein git-deps`, (c) do `lein install`. Currently, I have to manually do (b) and (c).

**Comment by torsten** [comment]
Hi, I had the same issue and tried to fix it in #7 .

If that PR is accepted, even `lein repl` should automatically pull the git dependencies and all of it's dependencies.

**Comment by callsSolve** [comment]
Hi , i keep on getting this error mesages :

Run lein deps
  lein deps
  shell: /bin/bash -e {0}
